### PR TITLE
[DOCS] Adds overview and API ref for cluster voting configurations

### PR DESCRIPTION
**Pull request by lcawl**
Related to https://github.com/elastic/elasticsearch/pull/34714

Adds an overview page for cluster voting configurations and a reference page for the API that manages the exclusion list.

### CODE REVIEW

**Comment by DaveCTurner** [comment]
@lcawl this looks like a good direction. I made a few changes and pushed 5b608a9 to your branch as it was tricky to express these suggestions in a review. Please feel free to revert or merge that commit as you see fit.

File: docs/reference/cluster/voting-exclusions.asciidoc [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+[float]
+=== Path parameters
+
+`node_name`::
+  A <<cluster-nodes,node filter>> that identifies {es} nodes.
+
+[float]
+=== Description
+  
+If the <<modules-discovery-settings,`cluster.auto_shrink_voting_configuration`
```
Comment (DaveCTurner) [inline docs/reference/cluster/voting-exclusions.asciidoc]: I'm concerned that mentioning this setting so prominently here will encourage too many users to fiddle with it. It should be left set to the default of `true` in the overwhelming majority of clusters. We need it to be configurable to support some very unusual resilience requirements, but it's only part of the story for those cases.

I think it'd be better to word most of this description as if the setting is set to the default of `true`.

Comment (lcawl) [reply docs/reference/cluster/voting-exclusions.asciidoc]: Thanks for the feedback!  I've updated the API description to downplay that setting.

**Review by DaveCTurner — APPROVED** [review]
Looks great, thanks.

**Pull request merged**